### ISSUE TITLE
Horizon Site Editor: Not Loading

### ISSUE THREAD

**Issue by jeyip** [Automattic/wp-calypso#47931]
#### Steps to reproduce

1. Starting at URL: https://horizon.wordpress.com/new?flags=gutenboarding/site-editor
2. Step through the horizon flow and create a site editor enabled site
3. Select either theme
4. Verify that, when dropped into the site editor, content does not load

#### What I expected

Block editor content loads automatically in the site editor

#### Screenshot / Video
![image (1)](https://user-images.githubusercontent.com/5414230/100786495-193f0100-33c7-11eb-8e5c-40a5e5c4e35e.png)

**Comment by ianstewart** [comment]
I can reproduce this outside of the horizon context by activating both a block based theme and the site editor via sticker from the command line as well.

1. Activate site editor via sticker from command line.
2. Activate site editor experiment via sticker from the command line.
3. Visit site editor, no content is loaded.

**Comment by jeyip** [comment]
It appears as if the correct template exists in the database. Logging information about the loaded template from `template-loader.php` returns post content that I'd expect. 🤔 

```
// Post Content for the front page wp_template CPT that's loaded in the site editor
<!-- wp:template-part {"slug":"header","theme":"twentytwentyone-blocks","align":"full","tagName":"header","className":"site-header"} /-->
<!-- wp:post-content /-->
<!-- wp:template-part {"slug":"footer","theme":"twentytwentyone-blocks","align":"full","tagName":"footer","className":"site-footer"} /-->
```

**Comment by jeyip** [comment]
Notes about what I researched yesterday:

It looks like, at the very least, when opening the site editor, post content for the front-page template has the correct markup. Here are the results from `wp-content/plugins/gutenberg-core/9.4.1/lib/template-loader.php`
when logging `$current_template`

```
// Post Content for the default template
<!-- wp:template-part {"slug":"header","theme":"twentytwentyone-blocks","align":"full","tagName":"header","className":"site-header"} /-->
<!-- wp:post-content /-->
<!-- wp:template-part {"slug":"footer","theme":"twentytwentyone-blocks","align":"full","tagName":"footer","className":"site-footer"} /-->
```

**Comment by jeyip** [comment]
~I can reproduce a similar issue locally now.~

~1. Spin up a local instance of Gutenberg and the site editor (happy to elaborate on this if people are unsure of how to do this)~
~2. Select a block based theme~
~3. Navigate to the site editor~
~4. Verify that no content loads~

~Unsure if it's caused by the same problem as what we're seeing with the horizon flow bug. Investigating further.~

Nevermind -- the site editor ended up rendering after an abnormally long waiting period 🐌

**Comment by jeyip** [comment]
Initiating a `WP_Query` in the sandbox returns what appear to be the correct CPT block templates from twenty-twenty-one-blocks. They're also auto-drafted.

```php
// Query for all template CPTs of type wp_template
// Placed in wp-content/plugins/gutenberg-core/9.4.1/lib/template-loader.php:171
$test_query = new WP_Query(
		array(
			'post_type'      => 'wp_template',
			'post_status'    => array( 'publish', 'auto-draft' ),
			'posts_per_page' => -1,
			'no_found_rows'  => true,
			'meta_key'       => 'theme',
			'meta_value'     => wp_get_theme()->get_stylesheet(),
		)
	      );

// Example return value
(
    [0] => WP_Post Object
        (
            [ID] => 30
            [post_author] => 69220074
            [post_date] => 2020-12-01 10:20:01
            [post_date_gmt] => 0000-00-00 00:00:00
            [post_content] => <!-- wp:template-part {"slug":"header","theme":"twentytwentyone-blocks","align":"full","tagName":"header","className":"site-header"} /-->

<!-- wp:post-title {"level":1,"align":"wide"} /-->

<!-- wp:separator {"align":"wide","className":"is-style-twentytwentyone-separator-thick"} -->
<hr class="wp-block-separator alignwide is-style-twentytwentyone-separator-thick" />
<!-- /wp:separator -->

<!-- wp:spacer {"height":70} -->
<div style="height:70px;" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:post-content /-->
<!-- wp:post-comments /-->

<!-- wp:template-part {"slug":"footer","theme":"twentytwentyone-blocks","align":"full","tagName":"footer","className":"site-footer"} /-->
            [post_title] => page
            [post_excerpt] =>
            [post_status] => auto-draft
            [comment_status] => closed
            etc...
        )
```

There's a `single`, `page`, `index`, and `front-page` block template, which matches the twenty twenty one block templates defined [here](https://github.com/WordPress/theme-experiments/tree/master/twentytwentyone-blocks/block-templates). It seems like templates _are_ being initialized and saved correctly, even if they're not being displayed in wp admin > appearance > templates

**Note:** It's the same with template parts. There is a header and footer that matches twenty twenty one blocks block template parts, but they're not being displayed.

**Comment by jeyip** [comment]
More notes:

Since `WP_Query` for template and template parts seem to imply that the auto-drafted CPTS are being created correctly, I explored possible client side issues further.

- I noticed that, when inserting a template part block to the post editor in a site created by the horizon flow, clicking the "New Template Part" button causes a runtime exception. 
- `Cannot read property 'id' of undefined`. The error can be traced back to [this line](https://github.com/WordPress/gutenberg/blob/d7e7561b7ea8128766f4f9f4150dc7c039c2cdeb/packages/block-library/src/template-part/edit/placeholder/index.js#L36s://github.com/WordPress/gutenberg/blob/d7e7561b7ea8128766f4f9f4150dc7c039c2cdeb/packages/block-library/src/template-part/edit/placeholder/index.js#L36)
- This seems to imply that saveEntityRecord isn't behaving as expected which means that there's a problem either with `@wordpress/data` or `@wordpress/api-fetch` (used to request data from the public api endpoints).
- This might also explain why the site editor page doesn't load, as it relies on `@wordpress/data` to and `@wordpress/api-fetch` to hydrate data

**Comment by jeyip** [comment]
Debugging more client side code in production shows that a few more methods exposed by `@wordpress/data` are returning undefined. I'm unsure why. To be more specific, the value returned from `getEntityRecord` from `packages/edit-site/src/components/editor/index.js:95`  is undefined 🤔 

I'm going to start digging into `@wordpress/data`

**Comment by jeyip** [comment]
Bug Fix D53874-code

**Comment by jeyip** [comment]
It looks like, with the release of Gutenberg v9.5.0, which seems like it'll be released in the next few days, they'll also be fixing the horizon flow issue that we see here.

See D53544-code

**Comment by jeyip** [comment]
Tested the horizon flow in production. This issue is fixed by the release of Gutenberg v9.5.1.
